src/Spice86.Core/Emulator/InterruptHandlers/Input/Keyboard/BiosKeyboardBuffer.cs

```diff
@@ -1,48 +1,48 @@
 ﻿namespace Spice86.Core.Emulator.InterruptHandlers.Input.Keyboard;
 
 using Spice86.Core.Emulator.Memory;
-using Spice86.Core.Emulator.ReverseEngineer.DataStructure;
+using Spice86.Core.Emulator.Memory.Indexable;
 
 /// <summary>
 /// This is a memory based FIFO Queue used to store key codes
+/// Data about buffer start, end and positions are stored in the Bios Data Area.
 /// </summary>
-public class BiosKeyboardBuffer : MemoryBasedDataStructure {
-    private const ushort End = 0x482;
-    private const ushort Head = 0x41A;
-    private const int InitialLength = 0x20;
-    private const ushort InitialStartAddress = 0x41E;
-    private const ushort Start = 0x480;
-    private const ushort Tail = 0x41C;
+public class BiosKeyboardBuffer {
+    private readonly Indexable _indexable;
+    private readonly BiosDataArea _biosDataArea;
 
-    public BiosKeyboardBuffer(Memory memory) : base(memory, 0) {
+    public BiosKeyboardBuffer(Indexable indexable, BiosDataArea biosDataArea) {
+        _indexable = indexable;
```

Comment:
Calling this indexable is a mistake I feel.
We should name things for what they are used for, not for what type they are.
This is often the same thing, but in this case it is used as the main Memory so it should be called `memory`.